ui/src/app/views/workflow/show/workflow.component.ts

```diff
@@ -125,36 +168,88 @@ export class WorkflowShowComponent implements AfterViewInit {
         });
 
         if (join.source_node_id) {
-            join.source_node_id.forEach( nodeID => {
-                this.g.setEdge('node-' + nodeID, 'join-' + join.id, {});
+            join.source_node_id.forEach(nodeID => {
+                this.createEdge('node-' + nodeID, 'join-' + join.id, {});
             });
         }
 
         if (join.triggers) {
             join.triggers.forEach(t => {
                 this.createNode(t.workflow_dest_node);
-                this.g.setEdge('join-' + join.id, 'node-' + t.workflow_dest_node.id, {id: 'trigger-' + t.id});
+                this.createEdge('join-' + join.id, 'node-' + t.workflow_dest_node.id, {id: 'trigger-' + t.id});
             });
         }
     }
 
     createNode(node: WorkflowNode): void {
-        let nodeComponentFactory = this.componentFactoryResolver.resolveComponentFactory(WorkflowNodeComponent);
-        let componentRef = nodeComponentFactory.create(this.svgContainer.parentInjector);
-        componentRef.instance.node = node;
+        let componentRef = this.createNodeComponent(node);
         this.svgContainer.insert(componentRef.hostView);
-
         this.g.setNode('node-' + node.id, {
             label: () => {
                 return componentRef.location.nativeElement;
             }
         });
-
         if (node.triggers) {
             node.triggers.forEach(t => {
                 this.createNode(t.workflow_dest_node);
-                this.g.setEdge('node-' + node.id, 'node-' + t.workflow_dest_node.id, {id: 'trigger-' + t.id});
+                this.createEdge('node-' + node.id, 'node-' + t.workflow_dest_node.id, {id: 'trigger-' + t.id});
+            });
+        }
+    }
+
+    createNodeComponent(node: WorkflowNode): ComponentRef<WorkflowNodeComponent> {
+        let nodeComponentFactory = this.componentFactoryResolver.resolveComponentFactory(WorkflowNodeComponent);
+        let componentRef = nodeComponentFactory.create(this.svgContainer.parentInjector);
+        componentRef.instance.node = node;
+        componentRef.instance.workflow = this.detailedWorkflow;
+        componentRef.instance.project = this.project;
+
+        return componentRef;
+    }
+
+    private openEditTriggerModal(parentID: string, childID: string) {
+        let pID = Number(parentID.replace('node-', ''));
+        let cID = Number(childID.replace('node-', ''));
+        let node = Workflow.getNodeByID(pID, this.detailedWorkflow);
+        if (node && node.triggers) {
+            node.triggers.forEach(t => {
+                if (t.workflow_dest_node_id === cID) {
+                    this.selectedNode = _.cloneDeep(node);
+                    this.selectedTrigger = _.cloneDeep(t);
+                    return;
```

Comment:
Use .some instead for this case maybe https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some?v=control